src/cdk/text-field/autosize.ts

```diff
@@ -35,20 +35,31 @@ import {fromEvent, Subject} from 'rxjs';
 })
 export class CdkTextareaAutosize implements AfterViewInit, DoCheck, OnDestroy {
   /** Keep track of the previous textarea value to avoid resizing when the value hasn't changed. */
-  private _previousValue: string;
+  private _previousValue?: string;
   private _initialHeight: string | null;
   private readonly _destroyed = new Subject<void>();
 
   private _minRows: number;
   private _maxRows: number;
   private _enabled: boolean = true;
 
+  /**
+   * Whether the minRows has changed since the last resize. If the minRows has decreased, the
+   * height of the textarea needs to be recomputed to reflect the new minimum. The maxHeight
+   * does not have the same problem because it does not affect the textarea's scrollHeight.
+   */
+  private _minRowsChanged = false;
+
   private _textareaElement: HTMLTextAreaElement;
 
   /** Minimum amount of rows in the textarea. */
   @Input('cdkAutosizeMinRows')
   get minRows(): number { return this._minRows; }
   set minRows(value: number) {
+    if (value !== this._minRows) {
+      this._minRowsChanged = true;
```

Comment:
Should be able to reduce this to `this._minRowsChanged = value !== this._minRows`. It's unlikely for this to be called quickly enough as to have the `_minRowsChanged` be overwritten before we've had a chance to resize.

Reply:
I think even better is to keep track of the number at the last resize rather than a boolean. That way if they change it to something new and then back to the original we can skip the work (I don't know if this will ever come up in real situations, but technically more correct)

Reply:
Done